home/management/commands/importgradedata.py

```diff
@@ -72,15 +78,31 @@ def parse_professor(self, name: str):
         if name is None or name == "":
             return None
 
-        names = name.strip().split(",")
-        lastname = names[0]
-        firstname = names[-1].strip().split()[0]
-        professors = Professor.unfiltered.filter(
-            Q(name__istartswith=firstname) & Q(name__iendswith=lastname)
-        )
-        if professors.exists():
+        name = name.strip()
+
+        # .first() is ok here because at most one record will be returned
+        alias = ProfessorAlias.objects.filter(alias=name).first()
+        if alias:
+            return alias.professor
```

Comment:
umd hands us the data in the format `lastname, firstname morenames morenames`, which the previous code accounted for and this code does not. I don't think we want to be creating professors with a name of `Walsh, Shane Bolles` - it should be `Shane Bolles Walsh`.